### ISSUE TITLE
[BUG] Custom optimizer callable that needs keyword arguments

### ISSUE THREAD

**Issue by gugarosa** [microsoft/DeepSpeed#3560]
**Describe the bug**

In the `DeepSpeedEngine` class, on the `_configure_optimizer()` method, there is a condition to check whether the incoming `client_optimizer` is an instance of supported optimizers (for example `torch.optim.Optimizer`) or a callable function.

If I were to pass a callable, which is called by this line:

```
basic_optimizer = client_optimizer(model_parameters)
```

How would I pass the extra keyword arguments of the optimizer? Ideally, my goal is to let DeepSpeed attach the model's parameters to the optimizer and only pass the non-instantiated custom optimizer with the proper hyperparameters (learning rate, weight decay, etc) from outside.

Thanks for your attention and best regards,
Gustavo.

**Comment by jomayeri** [comment]
@gugarosa Please try the PR attached to this issue. You will be able to pass the parameters to the callable via the ds_config optimizer section as show [here](https://www.deepspeed.ai/docs/config-json/#optimizer-parameters). Please note, even with the callable the config still needs a "type" specified. Thanks.

**Comment by jomayeri** [comment]
@gugarosa I take it back there is an easier way without needing to modify DeepSpeed. If you create a wrapper for your optimizer and pass that to `deepspeed.initialize()` you will get the desired behavior. Here is an example

```Python
def optimizer_callable(params):
    opt = my_optimizer(params=params, lr=..., ... )
    return opt


model, opt, _, _ = deepspeed.initialize(model=model,
                                        optimizer=opt_callable,
                                        config=config)
```
